routes/form_action.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express');
+const utils = require('../helpers/utils');
+const router = express.Router();
+
+async function action(page, request) {
+    const inputMapping = request.body.inputMapping;
+    const submitButton = request.body.submitButton || null;
+
+    for (const [selector, params] of Object.entries(inputMapping)) {
+        const value = params.value || "no value was provided";
```

Comment:
No need in "no value was provided". It would be better to debug if it failed with logs, rather than with "No such name" or smth like this.

Reply:
I decided to leave an empty string. It will be clear during use that something is wrong if the required parameter is not provided.